### PR TITLE
Add migration guide for v9

### PR DESCRIPTION
**Pull request by connor-baer**
## Approach and changes

- Write the migration guide to document the major changes when upgrading from v8 to v9

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Review by sirineJ — COMMENTED** [review]
👏 👏 👏 
only left some minor suggestions

File: MIGRATION.md [hunk 1]
```diff
@@ -2,12 +2,165 @@
...
+### Renamed package scope
+
+The packages have moved from the `@sumup` to the `@sumup-oss` scope to avoid conflicts with private packages. To get started, remove the old design system packages, then install the new ones:
+
+```sh
+npm uninstall @sumup/circuit-ui @sumup/design-tokens @sumup/icons @sumup/intl
+npm install @sumup-oss/circuit-ui @sumup-oss/design-tokens @sumup-oss/icons @sumup-oss/intl temporal-polyfill
+```
+
+Note that [`temporal-polyfill`](https://www.npmjs.com/package/temporal-polyfill) is a new required peer dependency and that the [`@sumup-oss/intl`](https://www.npmjs.com/package/@sumup-oss/intl) peer dependency has been upgraded to v3. If your app also depends on `@sumup-oss/intl` (previously called `@sumup/intl`), you need to upgrade it as well. Refer to its [changelog](https://github.com/sumup-oss/intl-js/blob/main/CHANGELOG.md) for migration instructions.
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply MIGRATION.md:30]: I contemplated that as well, but came to the conclusion that it's different from the listed prerequisites in that it's not something that can or should be done _before_ upgrading to Circuit UI v9. Installing `temporal-polyfill` and upgrading `@sumup-oss/intl` v3 has to happen _alongside_ the upgrade, so I felt that it fit better in this section.

File: MIGRATION.md [hunk 2]
```diff
@@ -2,12 +2,165 @@
...
+
+### New typeface
+
+The default typeface has changed from [Aktiv Grotesk](https://www.daltonmaag.com/font-library/aktiv-grotesk.html) to [Inter](https://rsms.me/inter/), a variable font. Variable fonts combine a continuous range of weights and other "axes" into a single file. This speeds up page load times and enables more creative freedom. Inter is a close match to Aktiv Grotesk, so users shouldn't notice a difference.
+
+Circuit UI no longer loads the fonts by default. Instead, centrally import the new `@sumup-oss/design-tokens/fonts.css` file containing the `@font-face` declarations to load the Inter font family. For more details, refer to the documentation on [how to load fonts in your application](https://github.com/sumup-oss/circuit-ui/tree/main/packages/design-tokens#fonts).
+
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply MIGRATION.md:68]: I'll copy over the full instructions from the README to ensure developers don't miss preloading the fonts.

File: MIGRATION.md [hunk 3]
```diff
@@ -2,12 +2,165 @@
...
+- import { Calendar, type CalendarProps, type PlainDateRange } from '@sumup-oss/circuit-ui/experimental';
++ import { Calendar, type CalendarProps, type PlainDateRange } from '@sumup-oss/circuit-ui';
+```
+
+### Other changes
+
+- Changed the `PlainDateRange` type from an array to an object with start and end properties. This affects the Calendar component's `selection` prop. Use the new `updatePlainDateRange` helper function to update a date range when a user selects a date.
+- Deprecated the `InputElement` interface and narrowed the Input component's element type to `HTMLInputElement` and the TextArea component's element type to `HTMLTextAreaElement`. This affects `ref`s and event handlers.
+- Removed the Table component's deprecated `initialSortedRow` prop. Use the `initialSortedColumn` prop instead (🤖 `no-renamed-props`).
+
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply MIGRATION.md:163]: Nothing changed about the ImageInput label in v9. I added the `hideLabel` prop in v8.14, and João decided to keep the label hidden by default. There might be an obsolete changeset that we'll need to clean up.

File: MIGRATION.md [hunk 4]
```diff
@@ -2,12 +2,165 @@
...
+### New typeface
+
+The default typeface has changed from [Aktiv Grotesk](https://www.daltonmaag.com/font-library/aktiv-grotesk.html) to [Inter](https://rsms.me/inter/), a variable font. Variable fonts combine a continuous range of weights and other "axes" into a single file. This speeds up page load times and enables more creative freedom. Inter is a close match to Aktiv Grotesk, so users shouldn't notice a difference.
+
+Circuit UI no longer loads the fonts by default. Instead, centrally import the new `@sumup-oss/design-tokens/fonts.css` file containing the `@font-face` declarations to load the Inter font family. For more details, refer to the documentation on [how to load fonts in your application](https://github.com/sumup-oss/circuit-ui/tree/main/packages/design-tokens#fonts).
+
+### Typography APIs
+
+The typography components have been redesigned to improve visual hierarchy, ensure consistency across platforms and provide more flexible APIs for developers. The changes are fully backward compatible and can be adopted gradually.
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply MIGRATION.md:71]: Hm, that would change the meaning in a confusing way. I meant it as "The typography components have been redesigned to achieve 1, 2 and 3", not "The typography components have been redesigned to do 1, achieving 2 and 3".

**Pull request merged**